### PR TITLE
[feature/PLAYER-3147] implement new label functionality

### PR DESCRIPTION
**Pull request by yaroslavgribov**
Rework multi audio language labels.
Basically we had to reimplement label assignment based on new acceptance criteria.

### Implements 
 - [PLAYER-3147](https://jira.corp.ooyala.com:8443/browse/PLAYER-3147)

### Tests
 - All unit tests passed
 - Covered all acceptance criteria from [PLAYER-3147](https://jira.corp.ooyala.com:8443/browse/PLAYER-3147) description and corner cases

### CODE REVIEW

File: js/components/closed-caption-multi-audio-menu/helpers.js [hunk 1]
```diff
@@ -90,13 +88,75 @@ function getDisplayTitle(language, label) {
...
+        };
+
+        return transformedTrack;
+      });
+    } else {
+      var uniqueTracks = groupedKeys.map(function(key) {
+        // if there are multiple tracks with the same language code
+        if (groupedTracks[key].length > 1) {
+          // get each list of duplicating tracks
+          return groupedTracks[key].map(function(audioTrack) {
```
Comment (jpfranco) [inline js/components/closed-caption-multi-audio-menu/helpers.js]: It looks like we could possibly move this mapping to its own function, since it's used several times.

File: tests/components/closed-caption-multi-audio-menu/helpers-test.js [hunk 2]
```diff
@@ -77,6 +77,138 @@ describe('closed caption & multi-audio helpers', function() {
...
+
+      expect(transformedTracksList).toEqual(expectedTracksList);
+    });
+
+    it('should return tracks list with labels if tracks are not distinct', function() {
+      var tracksList = [
+        { id: 1, language: 'English', label: 'discard this', enabled: true },
+        { id: 2, language: 'English', label: 'discard this', enabled: false },
+        { id: 3, language: 'Spanish', label: 'discard this', enabled: false },
+        { id: 4, language: 'Spanish', label: 'discard this', enabled: false }
```
Comment (jpfranco) [inline tests/components/closed-caption-multi-audio-menu/helpers-test.js]: I think we could improve this test slightly by using different labels here.

**Comment by pilievOoyala** [comment]
LGTM. Going to merge and you can address the minor issues in another PR.
